dsws_client/client.py

```diff
@@ -136,7 +172,7 @@ def fetch_timeseries_data(  # noqa: PLR0913
         data_responses = itertools.chain.from_iterable(
             response.data_responses for response in responses
         )
-        return responses_to_records(data_responses)
+        yield from responses_to_records(data_responses)
```

Comment:
why don't you let `responses_to_records` iterate over responses, use `data_responses` and yield there?

Reply:
Not sure what you mean, that's what I'm doing no? I suppose we can just return the iterator instead of yielding from it though.